p2p/eth.py

```diff
@@ -40,7 +41,7 @@ class NewBlockHashes(Command):
 
 class Transactions(Command):
     _cmd_id = 2
-    structure = sedes.CountableList(P2PTransaction)
+    structure = sedes.CountableList(BaseTransactionCommonFields)
```

Comment:
They are related (because I need `hash()` on the `P2PTransaction` and getting rid of that `P2PTransaction` alltogether seems like a good thing to me. But I'm happy to move that into a seperate commit to make the core of the tx pool change stand out better.